src/io/XTcpSocket.h

```diff
@@ -0,0 +1,16 @@
+
+
+class XTcpSocket {
```

Comment:
Since we're in a Qt world, i think it would be good to make this a Qt object with signals + slots. As a motivating usecase, a new signal could be created to emit error strings. Furthermore, i think it would be good to modify the interface to be Qt centric (this is both a blessing and a curse of using Qt as a "do-everything framework", which Qt apps tend to do/be forced to).
```
class XTcpSocket : public QObject {
    Q_OBJECT
public:
    XTcpSocket() { sockfd = -1; };
    void close();
    void abort();
    int write(const QByteArray& buff, const int size);
    int read(QByteArray& buff, int size);
    int isOpen(void) { return (sockfd >= 0); };
    int connectToHost(const QString& host, int port);
    int socketDescriptor(void) { return sockfd; };

signals:
    void error(const QString& msg);

private:
    int sockfd;
};

Reply:
When I changed the XTcpSocket to inherit the QObject the tst_risc test stopped compiling due to lack of references. As I couldn't solve the problem, I didn't inherit the QObject.